specification.md

```diff
@@ -4175,27 +4155,61 @@ eventRef:
 
 </details>
 
-References a `produced` and `consumed` [event definitions](#Event-Definition) via the `produceEventRef` and `consumeEventRef` properties, respectively.
+Publish an [event definition](#Event-Definition) referenced via the `name` property.
 
 The `data` property can have two types: string or object. If it is of string type, it is an expression that can select parts of state data
 to be used as payload of the event referenced by `produceEventRef`. If it is of object type, you can define a custom object to be the event payload.
 
 The `contextAttributes` property allows you to add one or more [extension context attributes](https://github.com/cloudevents/spec/blob/main/cloudevents/spec.md#extension-context-attributes)
 to the trigger/produced event.
 
-The `consumeEventTimeout` property defines the maximum amount of time (ISO 8601 format literal or expression) to wait for the result event. If not defined it should default to the  [actionExecutionTimeout](#Workflow-Timeout-Definition).
-If the event defined by the `consumeEventRef` property is not received in that set time, action invocation should raise an error
-that can be handled in the states `onErrors` definition. In case the `consumeEventRef` is not defined, the `consumeEventTimeout` property is ignored.
+##### ConsumeEventRef Definition
 
-The `invoke` property defines how the function is invoked (sync or async). Default value of this property is
-`sync`, meaning that workflow execution should wait until the function completes (the result event is received).
-If set to `async`, workflow execution should just produce the trigger event and should not wait for the result event.
-Note that in this case the action does not produce any results (payload of the result event) and the associated actions eventDataFilter as well as
-its retry definition, if defined, should be ignored.
-Functions that are invoked via events (sync or async) do not propagate their errors to the associated action definition and the
-workflow state, meaning that any errors that happen during their execution cannot be handled in the workflow states
-onErrors definition. Note that errors raised during functions that are invoked sync or async in this case
-should not fail workflow execution.
+Wait for an event to arrive. 
+
+| Parameter | Description | Type | Required |
+| --- | --- | --- | --- |
+| [name](#Event-Definition) | Reference to the unique name of an event definition. Must follow the [Serverless Workflow Naming Convention](#naming-convention) | string | yes |
```

Comment:
Should we use eventRef rather than name?

Reply:
In the case we use the approach suggested by https://github.com/serverlessworkflow/specification/pull/826#discussion_r1517565305, my personal preference goes to `name`. Otherwise, I would go for `to` for for subscribe, and `event` in case of produce.

So:

```yaml
actions:

  - name: my-action-1
    subscribe:
      to: my-consumed-event
      
  - name: my-action-2
    publish:
      event: my-produced-event
```

Reply:
hmmm, you subscribe to an event and publish an event, so the "to" looks weird to me. The preposition is not really relevant, which is relevant is that you are refererring to an event that is defined in the header of the workflow.

Reply:
> hmmm, you subscribe to an event and publish an event, so the "to" looks weird to me

Yeah, you are right. If only we could use a similar terminology than `to` for `publish`, it would trully be awesome.

> you are refererring to an event that is defined in the header of the workflow

Yup, and that's another problem, thus my concern with the proposed terminology.
Ideally, we should not force the user to declare an event at top level. For the sake of readability, one should be able to inline the definition of a non-reused event. In most cases, you are not gonna "reuse" events anyways.

Reply:
If we adop a fluent syntax, _publish_ and _subsribe_ are the verbs, while _event_ (or _name_, but _event_ has more meaning) is the object, _to_ (and _a_) is just a nexus with no meaning.
Even if we allow defining the event inline, event still makes sense, because it can be the ref to the event definition or the event definition itself.

Reply:
@cdavernas @fjtirado so:

``` yaml
actions:
  - name: my-action-1
    subscribe: my-consumed-event
     
  - name: my-action-2
    publish: my-produced-event
```

Reply:
Seems clear and straightforward to me.

Reply:
publish and subscribe are object
you need to write
```
publish: 
  event: my-produced-event
  data: .eventContent
subscribe:
  event: my-consumed-evemt
  timeout: 5PTS
```

Reply:
Then they should be string (ref) or object, imho.

Reply:
For subscribe maybe. For publish, I put data as required, because an cloud event with no data is pretty strange scenario (can be done by returning empty json, but I think it should be explicit)

Reply:
I have changed to subscribe and publish, and name to event.